### PR TITLE
[Impeller] dont allocate capture strings in release mode.

### PR DESCRIPTION
**Pull request by jonahwilliams**
I'm not exactly sure why, but this seems to stop the allocation of capture labels in release mode.

Fixes https://github.com/flutter/flutter/issues/138908

### CODE REVIEW

**Review by gaaclarke — APPROVED** [review]
You know what is happening, there must be a place where someone is calling these functions with a `char*` so even though it is a `const std::string&` it will create a `std::string` which copies the value so that it can pass a reference to that.  `std::string_view` will never copy.  You can use a `std::string_view` here.  It's probably faster than `const std::string_view&` considering it's so small and avoids another level of indirection.

Test exempt: No logical change, the effects will be measured in a benchmark.

**Review by bdero — APPROVED** [review]
LGTM

File: impeller/core/capture.cc [hunk 1]
```diff
@@ -117,16 +117,16 @@ void Capture::Rewind() {
...
-  type_name Capture::Add##pascal_name(const std::string& label,                \
-                                      type_name value,                         \
+  type_name Capture::Add##pascal_name(std::string_view label, type_name value, \
                                       CaptureProperty::Options options) {      \
     if (!active_) {                                                            \
       return value;                                                            \
     }                                                                          \
     FML_DCHECK(element_ != nullptr);                                           \
                                                                                \
+    std::string label_clone = std::string(label);                              \
```
Comment (stuartmorgan) [inline impeller/core/capture.cc:127]: Can we not push the use of string_view further into the stack so callers don't have to make copies?

File: impeller/core/capture.h [hunk 2]
```diff
@@ -235,13 +235,14 @@ class Capture {
...
 
-  inline Capture CreateChild(const std::string& label) {
+  inline Capture CreateChild(std::string_view label) {
 #ifdef IMPELLER_ENABLE_CAPTURE
     if (!active_) {
       return Capture();
     }
 
-    auto new_capture = Capture(label);
+    std::string label_copy = std::string(label);
```
Comment (stuartmorgan) [inline impeller/core/capture.h:244]: Same question here.

File: impeller/core/capture.cc (hunk 1 above)
Comment (jonahwilliams) [reply impeller/core/capture.cc:127]: We could. I decided to stop here because the rest of this is debug only is compiled out for profile/release modes.

**Comment by jonahwilliams** [comment]
The benchmark I use to measure these smaller optimziations is https://flutter-flutter-perf.skia.org/e/?queries=sub_result%3D90th_percentile_frame_rasterizer_time_millis%26sub_result%3Daverage_frame_rasterizer_time_millis%26test%3Dcomplex_layout_scroll_perf_ios__timeline_summary&selected=commit%3D38127%26name%3D%2Carch%3Dintel%2Cbranch%3Dmaster%2Cconfig%3Ddefault%2Cdevice_type%3DiPhone_11%2Cdevice_version%3Dnone%2Chost_type%3Dmac%2Csub_result%3Daverage_frame_rasterizer_time_millis%2Ctest%3Dcomplex_layout_scroll_perf_ios__timeline_summary%2C

Its probably too much noise for us to identify individual regressions, but by landing these sorts of improvements we've been able to gradually drive this down over time.

**Pull request merged**